#VRML_SIM V8.5 utf8
WorldInfo {
  basicTimeStep 8
}
Viewpoint {
  orientation 0.016694372 0.99985749 -0.0025111904 1.0409093
  position 0.97837823 0.77508546 0.94242292
  follow "NAO_H25_V50"
}
Background {
  skyColor [
    0.4 0.7 1
  ]
}
Ceiling {
  translation 0 3 0
  size 5 5
}
DirectionalLight {
  ambientIntensity 1
  direction -0.1 -0.5 -0.3
}
RectangleArena {
  floorSize 5 5
}
Nao {
  hidden position_0_0 6.4396e-11
  hidden position2_0_0 -2.95229e-06
  hidden position_0_1 0.404897
  hidden position2_0_1 1.22203e-05
  hidden position_0_2 0.400088
  hidden position2_0_2 5.71957e-07
  hidden position_0_3 8.94517e-06
  hidden position_0_4 2.05363e-05
  hidden linearVelocity_0 0.513926 -0.34738 0.00856242
  hidden angularVelocity_0 0.0516294 -0.00555528 0.148989
  hidden linearVelocity_12 0.493049 -0.351606 0.0151319
  hidden angularVelocity_12 0.0516299 -0.00555533 0.149001
  hidden position_19_0 -0.0481311
  hidden linearVelocity_19 0.503029 -0.350376 0.0115863
  hidden angularVelocity_19 0.0514298 -0.00572333 0.20205
  hidden position_20_0 5.79943e-07
  hidden linearVelocity_20 0.501534 -0.337845 0.0114671
  hidden angularVelocity_20 0.0514302 -0.0057236 0.202053
  hidden position_21_0 0.00430497
  hidden linearVelocity_21 0.499755 -0.326552 0.0115157
  hidden angularVelocity_21 0.0514301 -0.00572407 0.202056
  hidden position_22_0 2.65654e-05
  hidden position_22_1 -7.11808e-11
  hidden position_22_2 1.76101e-09
  hidden linearVelocity_22 0.497371 -0.312442 0.0116383
  hidden angularVelocity_22 0.0514303 -0.00572417 0.202058
  hidden position_23_1 5.05729e-10
  hidden linearVelocity_23 0.498982 -0.304561 0.0109641
  hidden angularVelocity_23 0.0514303 -0.00572417 0.202058
  hidden position_25_0 4.30849e-10
  hidden linearVelocity_25 0.50154 -0.303657 0.0103355
  hidden angularVelocity_25 0.0514303 -0.00572417 0.202058
  hidden linearVelocity_26 0.503747 -0.305232 0.00986799
  hidden angularVelocity_26 0.0514303 -0.00572417 0.202058
  hidden position_27_0 6.50227e-10
  hidden linearVelocity_27 0.499138 -0.303219 0.0109715
  hidden angularVelocity_27 0.0514303 -0.00572417 0.202058
  hidden position_28_0 4.31059e-10
  hidden linearVelocity_28 0.501696 -0.302315 0.0103429
  hidden angularVelocity_28 0.0514303 -0.00572417 0.202058
  hidden linearVelocity_29 0.503903 -0.303889 0.00987544
  hidden angularVelocity_29 0.0514303 -0.00572417 0.202058
  hidden position_30_0 1.87387e-10
  hidden linearVelocity_30 0.503042 -0.30718 0.0101386
  hidden angularVelocity_30 0.0514303 -0.00572417 0.202058
  hidden linearVelocity_31 0.501647 -0.304857 0.0103919
  hidden angularVelocity_31 0.0514303 -0.00572417 0.202058
  hidden position_32_0 0.0545944
  hidden linearVelocity_32 0.504135 -0.341693 0.0116374
  hidden angularVelocity_32 0.051627 -0.00555748 0.149795
  hidden position_33_0 -1.05111e-06
  hidden linearVelocity_33 0.50228 -0.330178 0.0115208
  hidden angularVelocity_33 0.0516274 -0.00555774 0.149798
  hidden position_34_0 -0.0754383
  hidden linearVelocity_34 0.500158 -0.321337 0.0115638
  hidden angularVelocity_34 0.0516273 -0.0055582 0.149801
  hidden position_35_0 -6.97531e-11
  hidden position_35_1 -6.98501e-11
  hidden position_35_2 1.78792e-09
  hidden linearVelocity_35 0.497334 -0.310969 0.0116758
  hidden angularVelocity_35 0.0516275 -0.00555832 0.149802
  hidden position_36_1 2.17833e-09
  hidden linearVelocity_36 0.497979 -0.304861 0.010993
  hidden angularVelocity_36 0.0516275 -0.00555832 0.149802
  hidden position_38_0 4.24917e-10
  hidden linearVelocity_38 0.499887 -0.303991 0.0103605
  hidden angularVelocity_38 0.0516275 -0.00555832 0.149802
  hidden linearVelocity_39 0.501694 -0.305052 0.00989161
  hidden angularVelocity_39 0.0516275 -0.00555832 0.149802
  hidden position_40_0 5.92658e-10
  hidden linearVelocity_40 0.498137 -0.303519 0.0110004
  hidden angularVelocity_40 0.0516275 -0.00555832 0.149802
  hidden position_41_0 4.24832e-10
  hidden linearVelocity_41 0.500045 -0.302649 0.010368
  hidden angularVelocity_41 0.0516275 -0.00555832 0.149802
  hidden linearVelocity_42 0.501852 -0.303711 0.00989905
  hidden angularVelocity_42 0.0516275 -0.00555832 0.149802
  hidden position_43_0 1.72862e-10
  hidden linearVelocity_43 0.501348 -0.306451 0.0101653
  hidden angularVelocity_43 0.0516275 -0.00555832 0.149802
  hidden linearVelocity_44 0.500116 -0.304758 0.0104185
  hidden angularVelocity_44 0.0516275 -0.00555832 0.149802
  hidden position_45_0 -1.56979e-07
  hidden linearVelocity_45 0.530363 -0.346105 0.00327135
  hidden angularVelocity_45 0.0516272 -0.0055163 0.148927
  hidden position_46_0 -0.907641
  hidden linearVelocity_46 0.535659 -0.346685 0.00162095
  hidden angularVelocity_46 0.0516457 -0.00550049 0.148886
  hidden position_47_0 0.575086
  hidden linearVelocity_47 0.548373 -0.322157 0.000166401
  hidden angularVelocity_47 0.0513966 -0.00509571 0.470593
  hidden position_48_0 1.24869e-06
  hidden linearVelocity_48 0.279084 -0.384109 -0.00372111
  hidden angularVelocity_48 0.0591961 -0.000600764 -5.93176
  hidden position_49_0 2.891e-07
  hidden linearVelocity_49 0.0019398 -0.410643 -0.00648876
  hidden angularVelocity_49 0.0591962 -0.000600843 -5.93176
  hidden linearVelocity_50 -0.20033 -0.59906 -0.00849441
  hidden angularVelocity_50 0.0591969 -0.000600822 -5.93176
  hidden translation_53 0.0198579 -0.00700005 -0.0412765
  hidden rotation_53 -2.92692e-08 -8.28337e-10 1 -3.14159
  hidden linearVelocity_53 -0.258041 -0.576439 -0.00907437
  hidden angularVelocity_53 0.0591969 -0.000600826 -5.93176
  hidden position_57_0 -1.27669e-06
  hidden linearVelocity_57 0.531247 -0.342313 0.00325844
  hidden angularVelocity_57 0.0516356 -0.00556303 0.148952
  hidden position_58_0 -0.911412
  hidden linearVelocity_58 0.537094 -0.341934 0.00157868
  hidden angularVelocity_58 0.0516234 -0.00555585 0.148927
  hidden position_59_0 0.575517
  hidden linearVelocity_59 0.549707 -0.317741 9.13097e-05
  hidden angularVelocity_59 0.0514568 -0.00507728 0.459582
  hidden position_60_0 1.25022e-06
  hidden linearVelocity_60 0.279744 -0.380938 -0.00400696
  hidden angularVelocity_60 0.0593054 -0.000600567 -5.94277
  hidden position_61_0 -1.45991e-07
  hidden linearVelocity_61 0.00215974 -0.408698 -0.0067836
  hidden angularVelocity_61 0.0593053 -0.000600428 -5.94277
  hidden linearVelocity_62 -0.199782 -0.597863 -0.00878928
  hidden angularVelocity_62 0.0593056 -0.000600209 -5.94277
  hidden translation_65 0.0198565 0.00700005 -0.041278
  hidden rotation_65 2.56377e-08 -6.82502e-10 1 -3.14159
  hidden linearVelocity_65 -0.257645 -0.575017 -0.00936953
  hidden angularVelocity_65 0.0593056 -0.000600213 -5.94277
  translation -0.736164 0.972927 -0.008005
  rotation 0.99772936 -0.046989017 0.048251017 -1.5718
  controller "my_controller"
  bodySlot [
    Receiver {
      name "Robot_Receiver"
    }
  ]
  leftFootSlot [
    InertialUnit {
    }
    Gyro {
      name ""
    }
  ]
  rightFootSlot [
    DEF Right_Foot_Gyro Gyro {
      rotation 1 0 0 1.5708
      name "Right_Foot_Gyro"
    }
    InertialUnit {
      name "Right_Foot_IU"
    }
  ]
  leftHandSlot [
    Transform {
      translation 0.06 0 0
      children [
        BallJoint {
          jointParameters BallJointParameters {
            springConstant 1
          }
          endPoint Connector {
            isLocked TRUE
            autoLock TRUE
            distanceTolerance 0.05
            axisTolerance 3.14159
            rotationTolerance 3.14159
            numberOfRotations 0
            snap FALSE
          }
        }
      ]
    }
  ]
  rightHandSlot [
    Transform {
      translation 0.06 0 0
      children [
        BallJoint {
          jointParameters BallJointParameters {
            springConstant 1
          }
          endPoint Connector {
            isLocked TRUE
            autoLock TRUE
            distanceTolerance 0.05
            axisTolerance 3.14159
            rotationTolerance 3.14159
            numberOfRotations 0
            snap FALSE
          }
        }
      ]
    }
  ]
}
FloorLight {
  translation -2.00175 -8.88178e-16 1.89667
}
FloorLight {
  translation 2.00366 2.22044e-15 -2.13347
}
DEF Swing Robot {
  translation 0 2.2 0
  rotation 1 0 0 -1.5708
  children [
    DEF RArm Transform {
      translation 0 0.25 -0.06
      children [
        HingeJoint {
          jointParameters HingeJointParameters {
            position 0.307398
            axis 0 1 0
            anchor 0 1 0
            minStop -3.1416
            maxStop 3.1416
            dampingConstant 1
          }
          endPoint Solid {
            translation -0.37391 0.0099843 -0.655628
            rotation -0.94379168 -0.23312892 0.23432492 -1.63233
            children [
              Transform {
                translation 0 -0.75 0
                children [
                  DEF SeatShape Solid {
                    translation 0 0 0.25
                    children [
                      DEF Mid_bar Transform {
                        translation 0.05 0.05 0
                        children [
                          SolidBox {
                            size 0.08 0.05 0.01
                            physics Physics {
                            }
                          }
                        ]
                      }
                      Emitter {
                        name "SeatEmitter"
                      }
                      DEF Seat_IU InertialUnit {
                        name "Seat_IU"
                      }
                      Gyro {
                        name "Seat_Gyro"
                      }
                      DEF Right_bar Transform {
                        translation 0 0.04 0.1
                        children [
                          SolidBox {
                            size 0.1 0.05 0.01
                            physics Physics {
                            }
                          }
                        ]
                      }
                      DEF Left_bar Transform {
                        translation 0 0.04 -0.1
                        children [
                          SolidBox {
                            size 0.1 0.05 0.01
                            physics Physics {
                            }
                          }
                        ]
                      }
                      DEF Front_bar Transform {
                        translation 0.065 0.075 0
                        children [
                          SolidBox {
                            size 0.06 0.01 0.5
                            physics Physics {
                            }
                          }
                        ]
                      }
                      DEF C1 Connector {
                        translation 0 0 0.25
                        rotation 1 0 0 -1.5708
                        isLocked TRUE
                        autoLock TRUE
                        distanceTolerance 0.05
                        numberOfRotations 0
                        snap FALSE
                      }
                      DEF SeatCon Connector {
                        translation 0 0 0.25
                        rotation 1 0 0 -1.5708
                        autoLock TRUE
                        numberOfRotations 0
                      }
                      Shape {
                        appearance Appearance {
                          material Material {
                          }
                        }
                        geometry DEF SSeat Box {
                          size 0.1 0.01 0.5
                        }
                      }
                    ]
                    boundingObject USE SSeat
                    physics Physics {
                    }
                  }
                ]
              }
              DEF RVBar Shape {
                appearance Appearance {
                  material Material {
                  }
                }
                geometry Cylinder {
                  height 1.5
                  radius 0.05
                }
              }
            ]
            boundingObject USE RVBar
            physics Physics {
              centerOfMass [
                0 0 0
              ]
            }
            linearVelocity 0.302865 -0.172108 -2.8138e-05
            angularVelocity 5.54583e-05 -3.12236e-05 0.431181
          }
        }
      ]
    }
    DEF LArm Transform {
      translation 0 -0.25 -0.06
      children [
        HingeJoint {
          jointParameters HingeJointParameters {
            position 0.307505
            axis 0 1 0
            anchor 0 1 0
            minStop -3.1416
            maxStop 3.1416
            dampingConstant 1
          }
          endPoint Solid {
            translation -0.380072 0.00213098 -0.668882
            rotation 0.94380386 0.23315997 -0.23424497 1.63301
            children [
              DEF C1 Connector {
                translation 0 -0.75 0
                rotation 1 0 0 1.5708
                isLocked TRUE
                autoLock TRUE
                distanceTolerance 0.05
                numberOfRotations 0
                snap FALSE
              }
              DEF SVBar2 Shape {
                appearance Appearance {
                  material Material {
                  }
                }
                geometry Cylinder {
                  height 1.5
                  radius 0.05
                }
              }
            ]
            boundingObject USE SVBar2
            physics Physics {
              centerOfMass [
                0 0 0
              ]
            }
            linearVelocity 0.288028 -0.164654 -3.72031e-05
            angularVelocity 5.90625e-05 -2.33178e-05 0.431222
          }
        }
      ]
    }
    DEF SHBarShape Shape {
      appearance DEF SwingApp Appearance {
        material Material {
          diffuseColor 0.7 0.4 0.8
        }
        texture ImageTexture {
        }
      }
      geometry DEF SHBar Cylinder {
        height 0.5
        radius 0.05
      }
    }
  ]
  name "Swing"
  boundingObject USE SHBarShape
  controller "swingcontroller"
  selfCollision TRUE
}
